AmplifyPlugins/Internal/Sources/InternalAWSPinpoint/Analytics/LocalStorage/AnalyticsEventSQLStorage.swift

```diff
@@ -142,12 +142,16 @@ class AnalyticsEventSQLStorage: AnalyticsEventStorage {
         ORDER BY timestamp ASC
         LIMIT ?
         """
-        let rows = try dbAdapter.executeQuery(queryStatement, [limit])
+        let rows = try dbAdapter.executeQuery(queryStatement, [limit]).makeIterator()
         var result = [PinpointEvent]()
-        for element in rows {
-            if let event = PinpointEvent.convertToEvent(element) {
-                result.append(event)
+        do {
+            while let element = try rows.failableNext() {
```

Comment:
Is there a scenario where the next row can fail, but subsequent rows can still be successfully retrieved?

If so, would it better (or even possible) to continue iterating through instead of bailing on the first error thrown?

Reply:
What's funny is that I was gonna suggest the opposite: If get get an error when attempting to retrieve a subsequent row, we should fail the whole operation instead of just returning what we have so far.

The way I see it is that we're explicitly running a SELECT query that we expect to contain the things we've asked, so just giving us back a subset of stuff just because the DB failed at some point sounds incorrect.

Reply:
Yeah, I think just propagating the error would be enough, since it will be caught and handled by the callers of this method:
- [`AWSPinpointAnalyticsPlugin.flushEvents()`](https://github.com/aws-amplify/amplify-swift/blob/main/AmplifyPlugins/Analytics/Sources/AWSPinpointAnalyticsPlugin/AWSPinpointAnalyticsPlugin%2BClientBehavior.swift#L103)
- [`AnalyticsClient`'s automatic flush of events](https://github.com/aws-amplify/amplify-swift/blob/main/AmplifyPlugins/Internal/Sources/InternalAWSPinpoint/Analytics/AnalyticsClient.swift#L297)